### ISSUE TITLE
lxd init create new btrfs storage pool on disk partition

### ISSUE THREAD

**Issue by danielecim** [lxc/lxd#6321]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 18.04
 * The output of "lxc info" or if that fails:
   * Kernel version: 5.0.0-31-generic
   * LXC version: 3.18
   * LXD version: 3.18
   * Storage backend in use: brtfs

# Issue description

Initialize lxd via lxd init with new storage pool (btrfs) on disk partition raise error. Example create new btrfs  storage pool on /dev/sda3  raise error, with disk device /dev/sdb it works. 

# Steps to reproduce

 1. sudo lxd init
 2. Dialog:
Would you like to use LXD clustering? (yes/no) [default=no]: 
Do you want to configure a new storage pool? (yes/no) [default=yes]: 
Name of the new storage pool [default=default]: 
Name of the storage backend to use (btrfs, ceph, dir, lvm, zfs) [default=zfs]: btrfs
Create a new BTRFS pool? (yes/no) [default=yes]: 
Would you like to use an existing block device? (yes/no) [default=no]: yes
**Path to the existing block device: /dev/sda3**
Would you like to connect to a MAAS server? (yes/no) [default=no]: 
Would you like to create a new local network bridge? (yes/no) [default=yes]: 
What should the new bridge be called? [default=lxdbr0]: 
What IPv4 address should be used? (CIDR subnet notation, “auto” or “none”) [default=auto]: 
What IPv6 address should be used? (CIDR subnet notation, “auto” or “none”) [default=auto]: 
Would you like LXD to be available over the network? (yes/no) [default=no]: 
Would you like stale cached images to be updated automatically? (yes/no) [default=yes] 
Would you like a YAML "lxd init" preseed to be printed? (yes/no) [default=no]: 

**Error: Failed to create storage pool 'default': Failed to create the BTRFS pool: btrfs-progs v4.4**
See http://btrfs.wiki.kernel.org for more information.

**Comment by danielecim** [comment]
Workaround:

sudo mkfs.btrfs /dev/sda3
sudo mkdir /mnt/lxdpool
sudo mount /dev/sda3 /mnt/lxdpool  or  edit /etc/fstab

sudo lxd init

Would you like to use LXD clustering? (yes/no) [default=no]: 
Do you want to configure a new storage pool? (yes/no) [default=yes]: 
Name of the new storage pool [default=default]: 
Name of the storage backend to use (btrfs, ceph, dir, lvm, zfs) [default=zfs]: btrfs
Create a new BTRFS pool? (yes/no) [default=yes]: no
**Name of the existing BTRFS pool or dataset: /mnt/lxdpool**
Would you like to connect to a MAAS server? (yes/no) [default=no]: 
Would you like to create a new local network bridge? (yes/no) [default=yes]: 
What should the new bridge be called? [default=lxdbr0]: 
What IPv4 address should be used? (CIDR subnet notation, “auto” or “none”) [default=auto]: 
What IPv6 address should be used? (CIDR subnet notation, “auto” or “none”) [default=auto]: 
Would you like LXD to be available over the network? (yes/no) [default=no]: 
Would you like stale cached images to be updated automatically? (yes/no) [default=yes] 
Would you like a YAML "lxd init" preseed to be printed? (yes/no) [default=no]:

**Comment by tomponline** [comment]
When this originally came up before, @stgraber suggested that a likely cause of this error was that there was an existing filesystem or partition on the device that was detected.

@danielecim the specific bug that results in the output of `btrfs-progs v4.4` only was fixed here https://github.com/lxc/lxd/pull/6282 so in the future if this occurs again you should see the actual error message.

**Comment by stgraber** [comment]
Just did some testing here, using the same snap and kernel version, a clean block device or partition does work fine, one which is dirty (has an existing label/filesystem) causes a failure.

Prior to the fix we did recently, this would result in an error matching the one in this issue, with the recent fix applied, I now get:
```
root@edfu:~# lxc storage create btrfs btrfs source=/dev/sdd
Error: Failed to create the BTRFS pool: Failed to run: mkfs.btrfs /dev/sdd -L btrfs: /dev/sdd appears to contain an existing filesystem (btrfs).
Use the -f option to force overwrite. (btrfs-progs v4.4
See http://btrfs.wiki.kernel.org for more information.
)
```

So it seems likely to be the source of the issue here and even if it's not, the direct error from mkfs.btrfs will now be visible. Closing since we do have a fix in master for this already.
